### ISSUE TITLE
No device list paging

### ISSUE THREAD

**Issue by Steve-Mcl** [flowforge/flowforge#2379]
Follow up to https://github.com/flowforge/security/issues/16

* Team Devices: Lack of paging causing device listing pages to slowdown when lots of devices exist
* Instance Devices: Lack of paging causing device listing pages to slowdown when lots of devices exist

_Originally posted by @Steve-Mcl in https://github.com/flowforge/security/issues/16#issuecomment-1614867214_

**Comment by MarianRaphael** [comment]
See also https://github.com/flowforge/forge-ui-components/issues/101

**Comment by Pezmc** [comment]
From my research so far, this isn't as straighforward as it might seem on the surface:
 - Both pages allow search and sort, search and sort is handled client side
 - Adding pagination means search/sort needs to be handled server side always (or a hybrid approach)
 - https://github.com/flowforge/forge-ui-components/issues/101 is tracking the UI component changes to support this (and https://github.com/flowforge/flowforge/issues/1160 for FlowForge).

So to summarise; to be able add pagination to devices, we first need to implement server side search/sort functionality to ui-components, then forge itself.

@MarianRaphael I'm resizing this to large, how should this be prioritised compared to other work in this release? Is a quick fix, like breaking search/sort for devices with pages acceptable for now?

**Comment by MarianRaphael** [comment]
Not part of 1.11 - The plan is to merge it this afternoon (after the release) to main / staging to verify it and release it next week.

**Comment by MarianRaphael** [comment]
@Pezmc, I can't verify the described behavior on staging. See the screenshot below:
![Screenshot 2023-09-14 at 16 25 54](https://github.com/flowforge/flowforge/assets/73583313/c15a716b-1b41-49e0-ad12-8de8e8f242d6)
Any idea why this might be the case?

**Comment by Pezmc** [comment]
@MarianRaphael What are you expecting to see here? You only have 18 devices, so there's no need to paginate. The threshold is set at `100`

**Comment by MarianRaphael** [comment]
Ah, okay. This is good to know! I thought the threshold is lower. All good in this case.

**Comment by Pezmc** [comment]
100 was set based on @Steve-Mcl's research in https://github.com/flowforge/security/issues/16#issuecomment-1614860247. It can be easily tweaked down the line however!
